src/components/Form/Integer.js

```diff
@@ -48,6 +48,17 @@ const fixInt = (value, oldValue, keyCode) => {
   return '0';
 };
 
+const isInputValid = (allowedKeys, keyCode, targetValue, validRegex) => {
+  if (allowedKeys.includes(keyCode)) {
```

Comment:
Please let's discuss this offline.
If the semantics is as expected, this function can be simplified.
If not, the overall validation flow needs to be explained, seems to be over-complicated and possible hard to unit test.